Sources/Data/Microsoft.Psi.Data/Dataset.cs

```diff
@@ -26,22 +26,55 @@ public class Dataset
         /// </summary>
         public const string DefaultName = "Untitled Dataset";
 
+        private string name;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Dataset"/> class.
         /// </summary>
         /// <param name="name">The name of the new dataset. Default is <see cref="DefaultName"/>.</param>
+        /// <param name="savePath">The path to save the dataset (optional, default is empty).<see cref="DefaultName"/>.</param>
+        /// <param name="autoSaveOnChange">Whether the dataset automatically autosave changes if a path is given (optional, default is false).</param>
         [JsonConstructor]
-        public Dataset(string name = Dataset.DefaultName)
+        public Dataset(string name = Dataset.DefaultName, string savePath = "", bool autoSaveOnChange = false)
         {
             this.Name = name;
+            this.CurrentSavePath = savePath;
+            this.AutoSaveChanges = autoSaveOnChange;

```

Comment:
I would suggest naming these members this.Filename and this.AutoSave, for simplicity and consistence with other API call parameters. Also, let's make the parameter names match the member names, e.g. autoSaveOnChange -> autoSave